project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -55,20 +60,31 @@
      *     for places or for places details
      */
     public ImmutableList<Place> fetch(UserPreferences preferences) throws FetcherException {
-        TextSearchRequest query =
-            PlacesApi.textSearchQuery(
-                CONTEXT, createCuisinesQuery(preferences.cuisines()), preferences.location())
-                .radius(SEARCH_RADIUS)
+        PlacesSearchResult[] placesSearchResult;
+        int radiusMult = 1;
+        do {
+            try {
+                placesSearchResult = getPlacesSearchResults(
+                    genTextSearchRequest(preferences, INIT_SEARCH_RADIUS * radiusMult));
+            } catch (ApiException | InterruptedException | IOException e) {
+                throw new FetcherException("Couldn't fetch places from Places API", e);
+            }
+            radiusMult++;
+        } while (
+            placesSearchResult.length < MIN_NUM_OF_RESULTS && radiusMult <= MAX_RADIUS_MULT);
+        return createPlacesList(placesSearchResult);
+    }
+
+    private TextSearchRequest genTextSearchRequest(UserPreferences preferences, int radius) {
+        TextSearchRequest request = PlacesApi.textSearchQuery(
+            CONTEXT, createCuisinesQuery(preferences.cuisines()), preferences.location())
+                .radius(radius)
                 .maxPrice(PriceLevel.values()[preferences.maxPriceLevel()])
                 .type(TYPE);
         if (preferences.openNow()) {
-            query.openNow(preferences.openNow());
-        }
-        try {
-            return createPlacesList(getPlacesSearchResults(query));
-        } catch (ApiException | InterruptedException | IOException e) {
-            throw new FetcherException("Couldn't fetch places from Places API", e);
+            request.openNow(preferences.openNow());
         }
+        return request;
     }
```

Comment:
The whole change looks really good!!
It took me a little while to understand what was going on, and I think it might have been more readable for me if MAX_RADIUS_MULT was called something else, like MAX_NUM_ITERATIONS_OF_RADIUS_CHANGE (or something else in that field).

Reply:
Right! changed it